### PR TITLE
Makes sure there are never any out of bounds issues when accessing `data`

### PR DESCRIPTION
**Pull request by simonmitchell**

### CODE REVIEW

**Review by BenShutt — COMMENTED** [review]
All looks fine, could we reuse the code in some way

```
func row(for indexPath: IndexPath) -> Row? {
    guard indexPath.section < data.count else {
        return nil
    }
    
    let section = data[indexPath.section]
    guard indexPath.row < section.rows.count else {
        return nil
    }
    
    let row = section.rows[indexPath.row]
    return row
}
```

you could do things based on null and not null.

If you wanted to go one step further you could conform to `Collection` and use the `data[indexPath]` syntax to get a row.

**Comment by simonmitchell** [comment]
I was literally thinking the same thing just now @BenCShutt! I'm going to conform to `Collection` :D Such a nice swift way of doing it!

File: ThunderTable/TableViewController+Collection.swift [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        
+        let section = data[index.section]
+        guard index.row < section.rows.count else {
+            return nil
+        }
+        
+        return (section, section.rows[index.row])
+    }
+    
+    public subscript (row index: IndexPath) -> Row? {
```
Comment (BenShutt) [inline ThunderTable/TableViewController+Collection.swift:27]: Does return `self[indexPath].row` work?

File: ThunderTable/TableViewController+Collection.swift [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        
+        let section = data[index.section]
+        guard index.row < section.rows.count else {
+            return nil
+        }
+        
+        return section.rows[index.row]
+    }
+    
+    public subscript (section index: IndexPath) -> Section? {
```
Comment (BenShutt) [inline ThunderTable/TableViewController+Collection.swift:31]: I'd keep this one the same though as there's no need to waste time getting the row

File: ThunderTable/TableViewController+Collection.swift (hunk 1 above)
Comment (simonmitchell) [reply ThunderTable/TableViewController+Collection.swift:27]: Good shout again! such a code re-use guru ;)

**Review by BenShutt — APPROVED** [review]

**Pull request merged**